migrations/migration.py

```diff
@@ -0,0 +1,139 @@
+import os
+from typing import List
+from datetime import datetime
+
+from isacc_messaging.audit import audit_entry
+from isacc_messaging.models.fhir import (
+    HAPI_request,
+    first_in_bundle,
+    next_in_bundle,
+    resolve_reference,
+)
+from migrations.migration_resource import MigrationManager
+
+class Migration:
+    def __init__(self):
+        self.migrations_dir = os.path.join(os.path.dirname(__file__), "versions")
+        self.latest_applied_migration = self.get_applied_migration_from_fhir()
+
+    def get_migration_files(self) -> List[str]:
+        """Retrieve the list of migration files."""
+        migration_files = sorted(os.listdir(self.migrations_dir))
+        return migration_files
+
+    def get_latest_migration(self) -> int:
+        """Find the latest migration number."""
+        migration_files = self.get_migration_files()
+        if migration_files:
+            latest_migration = int(migration_files[-1].split("_")[0])
+            return latest_migration
+        else:
+            return 0
+
+    def get_applied_migration_from_fhir(self) -> int:
+        """Retrieve the latest applied migration number from FHIR."""
+        # Logic to retrieve the latest applied migration number from FHIR
+        basic = HAPI_request('GET', 'Basic', params={
+            "identifier": f"http://isacc.app/twilio-message-sid|{id}"
+        })
+        basic = first_in_bundle(basic)
+
+        if basic is None:
+            self.create_new_migration_manager()
+
+            return 1
```

Comment:
i'd return `None` here.  we haven't yet successfully upgraded to this version number you are returning, right?